docs/imaging/dlc.rst

```diff
@@ -0,0 +1,114 @@
+
+Future methods may be added, including potentially 3D information, but these methods must first be written.
+
+
+
+DLC Models
+--------------
+
+1. In DeepLabCut model directory hierarchy, accessing the model file requires:
+
+  * cfg (fields: Task | date | iteration)
+  * shuffle (not in config)
+  * trainingset_fraction - not in config, compute as: cfg['TrainingFraction'][trainingsetindex]
+  
+  (https://github.com/DeepLabCut/DeepLabCut/blob/326e6f7e48b70bee9360ba84a8ffc1772f50cbe3/deeplabcut/utils/auxiliaryfunctions.py#L451)
+
+2. Minimal setting for triggering DLC analysis is as follows:
+
+      ``deeplabcut.analyze_videos(cfg, video_paths, shuffle, trainingsetindex)``
+      
+  (https://github.com/DeepLabCut/DeepLabCut/blob/master/deeplabcut/pose_estimation_tensorflow/predict_videos.py)
+
+3. Generated Scorer/network name
+
+  * Take an example: "DLC_resnet50_mouse_openfieldJun30shuffle1_1030000"
+  * Format: ``DLC_`` + netname (``resnet50``) + cfg['Task'] (``mouse_openfield``) + cfg['date'] (``June30``) + shuffle + trainingsiterations
+  * Note that "trainingsiterations" here is not the same as "dlc_iteration" (which is more of a model version)
+  * "trainingsiterations" is inferred based on: cfg['snapshotindex']
+  * This scorer name is not sufficiently unique because it misses information about the dlc_iteration
+  
+  (https://github.com/DeepLabCut/DeepLabCut/blob/326e6f7e48b70bee9360ba84a8ffc1772f50cbe3/deeplabcut/utils/auxiliaryfunctions.py#L524)
+
+In summary, the combination of (dlc_task, dlc_date, dlc_iteration, dlc_shuffle, dlc_trainingsetindex, dlc_snapshotindex) is sufficient to uniquely specify one DLC model
+
+
+
+Workflow
+--------------
+
+DLC is not currently supported via the web gui, and requires handling via code. Sample notebooks for model ingestion are provided:
+  
+  1. `DLC model insertion <https://github.com/kavli-ntnu/dj-moser-imaging/blob/master/Helper_notebooks/DLC%20model%20insertion.ipynb>`_
+  2. `SessionDLC insertion <https://github.com/kavli-ntnu/dj-moser-imaging/blob/master/Helper_notebooks/Insert%20Session%20DLC.ipynb>`_
+
+When a new model, or a new iteration of a model, is to be used, it must be registered in the Datajoint pipeline (see notebook 1 above). When a Session will be processed via DLC, it should be added to the ``SessionDLC`` table, (see notebook 2 above).
+
+The Imaging pipeline supports automated DLC processing, assuming that the model requested is available to one or more of the workers running the pipeline. If the model is not available, the job will never complete.
+
+Alternatively, users may run the model themselves (just as with Suite2p), although in this case, several manual steps must also be taken
+
+  * TODO
+
+Filenaming for DeepLabCut files is discussed in :ref:`Imaging naming dlc`.
+
```

Comment:
Additionally, if manually processed (i.e. run outside the DLC pipeline and then ingested into it), the helper notebook "Add timestamp to DLC pickle.ipynb" helps to add a timestamp to the first pickle DLC file, which matches the timestamp that is recorded in miniscope scanimage tif files. This is needed because DLC does not save any system time, etc. information, but we want to fill this timestamp in the Tracking tables.